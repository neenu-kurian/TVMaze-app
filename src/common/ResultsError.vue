<template>
  <div class="error__wrapper flex-center">
    <div class="error__image">
         <img src="../assets/images/sad-face.jpeg" alt="sad-face" class="full-width"/>
    </div>
    <div class="error__text">
        <error-text v-if="!error">Sorry, we couldnt find any results. Please try another tv show</error-text>
        <error-text v-else>Something went wrong, please try again later</error-text>
      </div>
    </div>
</template>

<script>
import ErrorText from '@/common/ErrorText.vue'
import { mapState } from 'vuex'

export default {
  name: 'ResultsError',
  components: {
    ErrorText
  },
  computed: {
    ...mapState(['error'])
  }
}
</script>
