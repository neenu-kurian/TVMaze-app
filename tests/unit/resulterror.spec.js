import { createLocalVue, mount } from '@vue/test-utils'
import ResultsError from '@/common/ResultsError'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('ResultsError.vue', () => {
  it('displays error text correctly when api call is success', () => {
    const wrapper = mount(ResultsError, {
      computed: {
        error: () => false
      },
      slots: {
        default: 'Sorry, we couldnt find any results. Please try another tv show'
      },
      localVue
    })

    expect(wrapper.find('.error__message').text()).toContain('Sorry, we couldnt find any results. Please try another tv show')
  })

  it('displays error text correctly when api call is failed', () => {
    const wrapper = mount(ResultsError, {
      computed: {
        error: () => true
      },
      slots: {
        default: 'Something went wrong, please try again later'
      },
      localVue
    })

    expect(wrapper.find('.error__message').text()).toContain('Something went wrong, please try again later')
  })

  it('renders the component correctly', () => {
    const wrapper = mount(ResultsError, {
      computed: {
        error: () => false
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})
