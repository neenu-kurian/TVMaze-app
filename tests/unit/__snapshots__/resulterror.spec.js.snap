// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsError.vue renders the component correctly 1`] = `
<div class="error__wrapper flex-center">
  <div class="error__image"><img src="../assets/images/sad-face.jpeg" alt="sad-face" class="full-width"></div>
  <div class="error__text">
    <p class="error__message">Sorry, we couldnt find any results. Please try another tv show</p>
  </div>
</div>
`;
